src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -175,11 +232,83 @@ private SafeCurlHandle CreateRequestHandle(RequestCompletionSource state)
             }
 
-            // TODO: Handle headers, proxy and other options
+            SetProxyOptions(state, requestHandle);
+            // TODO: Handle headers and other options
 
             return requestHandle;
         }
 
+        private static void SetProxyOptions(RequestCompletionSource state, SafeCurlHandle requestHandle)
+        {
+            var requestUri = state.RequestMessage.RequestUri;
+            Debug.Assert(state.Handler != null);
+            if (state.Handler._proxyPolicy == ProxyUsePolicy.DoNotUseProxy)
+            {
+                Interop.libcurl.curl_easy_setopt(requestHandle, CURLoption.CURLOPT_PROXY, string.Empty);
+                return;
+            }
+
+            if ((state.Handler._proxyPolicy == ProxyUsePolicy.UseDefaultProxy) || (state.Handler.Proxy == null))
+            {
+                return;
+            }
+
+            Debug.Assert( (state.Handler.Proxy != null) && (state.Handler._proxyPolicy == ProxyUsePolicy.UseCustomProxy));
+            if (state.Handler.Proxy.IsBypassed(requestUri))
+            {
+                Interop.libcurl.curl_easy_setopt(requestHandle, CURLoption.CURLOPT_PROXY, string.Empty);
+                return;
+            }
+
+            var proxyUri = state.Handler.Proxy.GetProxy(requestUri);
+            if (proxyUri == null)
+            {
+                return;
+            }
+
+            Interop.libcurl.curl_easy_setopt(requestHandle, CURLoption.CURLOPT_PROXYTYPE, CURLProxyType.CURLPROXY_HTTP);
+            Interop.libcurl.curl_easy_setopt(requestHandle, CURLoption.CURLOPT_PROXY, proxyUri.AbsoluteUri);
+            Interop.libcurl.curl_easy_setopt(requestHandle, CURLoption.CURLOPT_PROXYPORT, proxyUri.Port);
+            NetworkCredential credentials = GetCredentials(state.Handler.Proxy.Credentials, requestUri);
+            if (credentials != null)
+            {
+                if (string.IsNullOrEmpty(credentials.UserName))
+                {
+                    throw new ArgumentException(SR.net_http_argument_empty_string, "UserName");
+                }
+
+                string credentialText;
+                if (string.IsNullOrEmpty(credentials.Domain))
+                {
+                    credentialText = string.Format("{0}:{1}", credentials.UserName, credentials.Password);
+                }
+                else
+                {
+                    credentialText = string.Format("{2}\\{0}:{1}", credentials.UserName, credentials.Password, credentials.Domain);
+                }
+                Interop.libcurl.curl_easy_setopt(requestHandle, CURLoption.CURLOPT_PROXYUSERPWD, credentialText);
+            }
+        }
+
+        private static NetworkCredential GetCredentials(ICredentials proxyCredentials, Uri requestUri)
+        {
+            if (proxyCredentials == null)
+            {
+                return null;
+            }
+
+            foreach (var authScheme in AuthenticationSchemes)
```

Comment:
Nit: var -> string.  This can be changed when you're making additional updates in your next PR.